libs/search-widget/src/widgets/row/Thumbnail.svelte

```diff
@@ -0,0 +1,37 @@
+<script lang="ts">
+  import { onDestroy } from 'svelte';
+  import { getFile } from '../../core/api';
+
+  export let src: string;
+  export let ratio: number = 5/4;
+
+  let thumbnail: string;
+  if (src) {
+    getFile(src).subscribe((url) => (thumbnail = url));
+  }
+  onDestroy(() => {
+    if (thumbnail) URL.revokeObjectURL(thumbnail);
+  });
+</script>
+
+<div class="thumbnail" style:padding-top="{`${Math.round(1/ratio*100)}%`}">
```

Comment:
I'm a bit surprised by this padding-top: does it mean we have more grey background on the top than on the bottom?